### PR TITLE
Fix: Deprecated warnings

### PR DESCRIPTION
**Pull request by lucas-zimerman**
This PR fixes some deprecated warnings from both Sentry Native and Capacitor dependency.

No break changes were introduced with this PR.

#skip-changelog

### CODE REVIEW

File: android/src/main/java/io/sentry/capacitor/SentryCapacitor.java [hunk 1]
```diff
@@ -208,7 +214,7 @@ public void fetchNativeRelease(PluginCall call) {
         JSObject release = new JSObject();
         release.put("id", this.packageInfo.packageName);
         release.put("version", this.packageInfo.versionName);
-        release.put("build", String.valueOf(this.packageInfo.versionCode));
+        release.put("build", String.valueOf(this.packageInfo.versionCode)); // Requires API 28
```
Comment (krystofwoldrich) [inline android/src/main/java/io/sentry/capacitor/SentryCapacitor.java:217]: Can we use this as the min API version is 22 in the SDK build.gradle

File: android/src/main/java/io/sentry/capacitor/SentryCapacitor.java [hunk 2]
```diff
@@ -180,17 +182,21 @@ public void setUser(PluginCall call) {
...
 
-                      otherUserKeysMap.put(key, value);
+                      // other is ConcurrentHashMap and can't have null values
+                      if (value != null) {
+                        otherUserKeys.put(key, value);
+                      }
                     }
 
-                    userInstance.setOthers(otherUserKeysMap);
+                    // Works on the Android SDK but doesn't seems to be supported by the back-end.
```
Comment (krystofwoldrich) [inline android/src/main/java/io/sentry/capacitor/SentryCapacitor.java:198]: Could you link and example event, it the data are in the processed event JSON, it's  good. Only UI issue.

File: android/src/main/java/io/sentry/capacitor/SentryCapacitor.java (hunk 1 above)
Comment (krystofwoldrich) [reply android/src/main/java/io/sentry/capacitor/SentryCapacitor.java:217]: NVM I misunderstand the comment.

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀

File: android/src/main/java/io/sentry/capacitor/SentryCapacitor.java (hunk 2 above)
Comment (lucas-zimerman) [reply android/src/main/java/io/sentry/capacitor/SentryCapacitor.java:198]: https://[us.sentry.io/api/0/projects/sentry-sdks/sentry-capacitor/events/c15bc11c29a842488d8891b37416a211/json/](https://us.sentry.io/api/0/projects/sentry-sdks/sentry-capacitor/events/c15bc11c29a842488d8891b37416a211/json/)

You can see on the json payload the user has `data":{"teeth":15}},`
where the event itself doesn't show this additional data on the user
https://sentry-sdks.sentry.io/issues/5061824231/events/351f306ce50c401694c2bb040b80546f/?project=5522756

**Pull request merged**